enclave-runtime/src/tls_ra/tls_ra_client.rs

```diff
@@ -222,10 +245,28 @@ pub(crate) fn request_state_provisioning_internal<StateAndKeySealer: SealStateAn
 
 fn tls_client_config<A: EnclaveAttestationOCallApi + 'static>(
 	sign_type: sgx_quote_sign_type_t,
+	quoting_enclave_target_info: &sgx_target_info_t,
+	quote_size: u32,
 	ocall_api: A,
 	skip_ra: bool,
 ) -> EnclaveResult<ClientConfig> {
-	let (key_der, cert_der) = create_ra_report_and_signature(sign_type, skip_ra)?;
+	#[cfg(not(feature = "dcap"))]
+	let (key_der, cert_der) = create_ra_report_and_signature(
+		skip_ra,
+		RemoteAttestationType::Epid,
```

Comment:
Can we not just make the `RemoteAttestationType` dependent on the feature?

Reply:
Currently I do not see how one can disambiguate between DCAP and EPID without the necessary feature flag, as for them to work the necessary feature needs to be activated. Or did you mean that this decision should be an argument of this function (and the caller's decision) rather than hardwired?

Reply:
No, what I meant was simply:

```rust
#[cfg(not(feature = "dcap"))]
let attestation_type = RemoteAttestationType::Epid;

#[cfg(feature = "dcap")]
let attestation_type = RemoteAttestationType::Dcap;
```

Reply:
Of course, much nicer, thanks :100: